# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/android/nsDragService.h
# Commit: beeda5ef3712
# Full Hash: beeda5ef3712b73a19c7a4ce08a52ab4dd08b802
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 20 - Move EndDragSession from nsIDragService to nsIDragSession r=gstoll,rkraesig,win-reviewers,geckoview-reviewers,m_kato
#   
#   This is a straightforward move.  It does take the liberty of breaking out an
#   EndDragSessionImpl method, which will be needed later.
#   
# ==============================================================================

diff -r 6c3ffc8c534c -r beeda5ef3712 widget/android/nsDragService.h
--- a/widget/android/nsDragService.h	Thu Jul 04 07:48:10 2024 +0000
+++ b/widget/android/nsDragService.h	Thu Jul 04 07:48:11 2024 +0000
@@ -34,6 +34,9 @@
 
   virtual bool MustUpdateDataTransfer(mozilla::EventMessage aMessage) override;
 
+  MOZ_CAN_RUN_SCRIPT nsresult EndDragSessionImpl(
+      bool aDoneDrag, uint32_t aKeyModifiers) override;
+
  protected:
   virtual ~nsDragSession() = default;
 
@@ -53,9 +56,6 @@
 
   NS_DECL_ISUPPORTS_INHERITED
 
-  MOZ_CAN_RUN_SCRIPT NS_IMETHOD EndDragSession(bool aDoneDrag,
-                                               uint32_t aKeyModifiers) override;
-
   static void SetDropData(
       mozilla::java::GeckoDragAndDrop::DropData::Param aDropData);
 